### ISSUE TITLE
Support for SSH tunneling ?

### ISSUE THREAD

**Issue by tusharmath** [OptimalBits/redbird#26]
Can we have SSH tunneling support as well? 
I am currently using redbird with - https://www.npmjs.com/package/tunnel-ssh

**Comment by gavinengel** [comment]
+1

**Comment by manast** [comment]
not sure I understand this use case well, could you elaborate more on it or provide an example?

**Comment by gavinengel** [comment]
For myself,  I'm looking for a http reverse proxy and a ssh reverse proxy, for my CoreOS Docker server.  

So it would be great if Redbird could help me in both ways.

**Comment by DerekTBrown** [comment]
I am using Red Bird for this exact purpose- seems to work fine.

**Comment by DerekTBrown** [comment]
https://github.com/containous/traefik

**Comment by DerekTBrown** [comment]
My guess is that I am working on an identical usecase- trying to allow users to SSH into a Docker container based on domain name (a SSH Reverse Proxy).  It doesnt seem like there is an existing solution, so I am going to be implementing this in Red Bird.

**Comment by manast** [comment]
@DerekTBrown aha, ok, I think I understand it now :)

**Comment by DerekTBrown** [comment]
Honestly, it doesnt look like RedBird is super conducive to this sort of implementation.  Nothing against RedBird, but it is designed as an HTTP Proxy alone.  I am working on a fork of RedBird which will implement all of RedBird's functionality, but in a more modular way.

**Comment by tcoats** [comment]
We have a clone of redbird called redwire: https://github.com/metocean/redwire

It supports TLS and TCP proxying in a modular way. Worth looking at even if you're doing your own thing.

**Comment by manast** [comment]
@DerekTBrown yes, the initial requirements for redbird where to be a very simple to setup reverse proxy for http. I will keep modularizing it overtime though according to my own needs.

**Comment by DerekTBrown** [comment]
@manast @tcoats having read over both projects, I think that they are both somewhat limited in their ability to become modular without breaking backwards compatibility.  I also think it is somewhat silly that you are both developing what are essentially identical pieces of software in parallel.

What I am going to try to do (as this is something I need for my production application anyway) is to pull together both your projects with a more modular construction, and include inbuilt support for a couple of features which are in the pipeline for RedBird/RedWire:

- Remove Static HTTP/S Bindings
- Configuration Hotloading
- Official Docker Image (w/EncloseJS)
- Let's Encrypt Support
- Multiple Resolvers (https://github.com/OptimalBits/redbird/issues/11)
- Wildcard / Multiple Routers (https://github.com/OptimalBits/redbird/pull/48)

It will not be syntactically the same, but will be substantially more modular in the long run.  It is going to take me a while to get to a stable version, but if you like, you can follow the project here:

https://github.com/DerekTBrown/redrouter

**Comment by DerekTBrown** [comment]
@manast @tcoats  I wrote the SSH Proxy component of my project just to demonstrate how the framework would work.  Would you be interested in me porting the HTTP/HTTPS bits from RedBird and RedRouter to my framework as the next step on the development roadmap?
